python/pyspark/ml/param/__init__.py

```diff
@@ -65,6 +80,144 @@ def __eq__(self, other):
             return False
 
 
+class TypeConverters(object):
+    """
+    .. note:: DeveloperApi
+
+    Factory methods for common type conversion functions for `Param.typeConverter`.
+
+    .. versionadded:: 2.0.0
+    """
+
+    @staticmethod
+    def _is_numeric(value):
+        vtype = type(value)
+        return vtype in [int, float, np.float64, np.int64] or vtype.__name__ == 'long'
+
+    @staticmethod
+    def _is_integer(value):
+        return TypeConverters._is_numeric(value) and float(value).is_integer()
+
+    @staticmethod
+    def _can_convert_to_list(value):
+        vtype = type(value)
+        return vtype in [list, np.ndarray, tuple, xrange, array.array] or isinstance(value, Vector)
+
+    @staticmethod
+    def _can_convert_to_string(value):
```

Comment:
I changed this to do "safe" unicode to str conversions. The way it was previously, a user could provide non-ascii characters in a string param and get a somewhat mysterious `UnicodeEncodeError`. This way, they should at least get an error message consistent with other TypeConverters. I appreciate feedback on this.

Reply:
@sethah @jkbradley sorry for a question on an old bit of code, but hope this is a quick one:
These converters don't allow None. Some parameters can be set to None. Should they actually all just return None if their value is None? any harm in that?